### DISCUSSION TITLE
Does the `SafeBuffer.Read` method change the current position?

### DISCUSSION THREAD

**Discussion by udaken** [dotnet/runtime#86752] [Q&A] [answered]
The Remarks section for the [`SafeBuffer.AcquirePointer`](https://learn.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.safebuffer.acquirepointer?view=net-7.0#remarks) method states the following:

> [AcquirePointer](https://learn.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.safebuffer.acquirepointer?view=net-7.0) calls the [SafeHandle.DangerousAddRef](https://learn.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.safehandle.dangerousaddref?view=net-7.0) method and exposes the pointer. Unlike the [Read](https://learn.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.safebuffer.read?view=net-7.0) method, it does not change the current position of the pointer.

However, `SafeBuffer.Read` does not have the similar remark.

When I prepared a test program to verify this, I did not observe any behavior that changes the current position.

Could this be a documentation bug?

```csharp
using System.IO.MemoryMappedFiles;

File.WriteAllBytes("a.bin", new byte[]{0x1, 0x02, 0x03});
using(var mmf =  System.IO.MemoryMappedFiles.MemoryMappedFile.CreateFromFile("a.bin"))
{
  using (var accessor = mmf.CreateViewAccessor())
  {
     Console.WriteLine( $"{accessor.SafeMemoryMappedViewHandle.Read<byte>(0):X}"); // 1
     Console.WriteLine( $"{accessor.SafeMemoryMappedViewHandle.Read<byte>(0):X}"); // 1
     Console.WriteLine( $"{accessor.SafeMemoryMappedViewHandle.Read<byte>(0):X}"); // 1
  }
}
```

**Comment by teo-tsirpanis** [comment, marked as answer]
The documentation seems wrong. `SafeBuffer` does not have a notion of "current position"; all its read/write methods take the offset as a parameter.
